js/data.js

```diff
@@ -7,13 +7,65 @@ let data = {
   nextEntryId: 1,
 };
 
-const previousData = localStorage.getItem('javascript-local-storage');
-
 window.addEventListener('beforeunload', (event) => {
   const dataModel = JSON.stringify(data);
-  localStorage.setItem('javascript-local-storage', dataModel)
+  localStorage.setItem('javascript-local-storage', dataModel);
+});
+
+document.addEventListener('DOMContentLoaded', () => {
+  const previousData = localStorage.getItem('javascript-local-storage');
+
+  if (previousData !== null) {
+    data = JSON.parse(previousData);
+    data.entries.forEach((entry) => {
+      renderEntry(entry);
+    });
+    viewSwap(data.view);
+      if (data.entries.length > 0) {
+        toggleNoEntries();
+      }
+  } else {
+    viewSwap(data.view)
+  }
 });
 
-if (previousData !== null) {
-  data = JSON.parse(previousData);
+function renderEntry(entry) {
+  const entryList = document.querySelector('.entry-list');
+  const newLi = document.createElement("li");
+
+  newLi.innerHTML = '<li class="row entry-container"> <div class="column-half"><img class="img-preview" src= "' + entry.url + '"/></div> <div class="column-half entry-info-container"><div class="entry-title">' + entry.title + '</div> <div class="entry-notes">' + entry.notes + '</div></div></li > ' ;
```

Comment:
avoid using `innerHTML` it would be better to create a function called `renderEntry` that will build the HTML nodes and append them to another HTML node that you have selected from the DOM.

review the DOM creation exercise for this